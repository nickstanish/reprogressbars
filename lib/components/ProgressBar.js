import React from 'react';
import PropTypes from 'prop-types';
import ProgressProvider from './ProgressProvider';
import Bar from './Bar';

function only(originalObject, keys) {
  var newObject = {};
  keys.forEach(function (key) {
    if (originalObject[key] !== undefined) {
      newObject[key] = originalObject[key];
    }
  });
  return newObject;
}

export default function ProgressBar(props) {
  var barProps = only(props, ['color', 'height', 'className', 'useBoxShadow']);
  return React.createElement(
    ProgressProvider,
    { isLoading: props.isLoading },
    React.createElement(Bar, barProps)
  );
}

ProgressBar.propTypes = {
  color: PropTypes.string,
  height: PropTypes.string,
  isLoading: PropTypes.bool,
  className: PropTypes.string,
  useBoxShadow: PropTypes.bool
};