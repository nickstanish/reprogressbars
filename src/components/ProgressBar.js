import React, { Component, PropTypes } from 'react';
import ProgressProvider from './ProgressProvider';
import Bar from './Bar';

function only(originalObject, keys) {
  const newObject = {};
  keys.forEach((key) => {
    if (originalObject[key] !== undefined) {
      newObject[key] = originalObject[key];
    }
  });
  return newObject;
}

export default class ProgressBar extends Component {
  static propTypes = {
    color: PropTypes.string,
    height: PropTypes.string,
    isLoading: PropTypes.bool,
    className: PropTypes.string,
    useBoxShadow: PropTypes.bool
  };

  render() {
    const barProps = only(this.props, ['color', 'height', 'className', 'useBoxShadow']);
    return (
      <ProgressProvider isLoading={this.props.isLoading}>
        <Bar {...barProps} />
      </ProgressProvider>
    );
  }
}
