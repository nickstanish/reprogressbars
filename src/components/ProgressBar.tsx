import { Bar, BarProps } from "./Bar";
import { withProgress, WithProgressProps } from "../hoc/withProgress";

export const ProgressBar = withProgress<BarProps & WithProgressProps>(Bar);
