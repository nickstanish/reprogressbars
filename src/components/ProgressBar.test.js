import React from 'react';
import { mount } from 'enzyme';

import ProgressBar from './ProgressBar';

describe('ProgressBar', () => {
  test('renders', () => {
    expect(mount(<ProgressBar />)).toMatchSnapshot();
  });
});
