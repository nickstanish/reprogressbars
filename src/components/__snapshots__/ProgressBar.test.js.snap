// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders 1`] = `
<ProgressBar>
  <ProgressProvider>
    <Bar
      color="#29d"
      height="2px"
      progress={
        Object {
          "active": false,
          "value": 0,
        }
      }
      useBoxShadow={true}
    >
      <div
        className="reprogressbar reprogressbar--inactive"
      />
    </Bar>
  </ProgressProvider>
</ProgressBar>
`;
